src/cli.rs

```diff
@@ -851,7 +856,35 @@ impl clap::Args for Exec {
                            fd -g 'test_*.py' -X vim\n\n  \
                        - Find all *.rs files and count the lines with \"wc -l ...\":\n\n      \
                            fd -e rs -X wc -l\
-                     "
+                    "
+                ),
+        )
+        .arg(
+            Arg::new("filter")
+                .action(ArgAction::Append)
+                .long("filter")
+                .short('f')
+                .num_args(1..)
+                .allow_hyphen_values(true)
+                .value_terminator(";")
+                .value_name("cmd")
+                .conflicts_with_all(["exec", "exec_batch", "list_details"])
```

Comment:
Why does this need to conflict? 

Logically, it makes sense to allow filtering items first, and then passing the results that succeed through to exec or exec-batch.

Reply:
I like this idea, I'll consider how we can make these commands compatible.